File: fixtures/rules/AR014/edge/hand_rolled_counter_retry.py
# A hand-rolled counter compared against a variable limit -- the frontend's documented, stated
# limitation (see retries.py's module docstring): not recognized as a retry construct at all, so
# no RetryPolicy is produced and this rule stays silent. Silence, not a guess.
def upload_with_retry(path, max_attempts):
    attempts = 0
    while attempts < max_attempts:
        try:
            do_upload(path)
            break
        except OSError:
            attempts += 1

File: fixtures/rules/AR014/edge/hand_rolled_counter_retry.js
// A hand-rolled counter compared against a VARIABLE limit (not a literal) -- not detected by this
// frontend's manual-loop heuristic (retries.py's own documented limitation, mirrored here).
// Silence, not a guess: no RetryPolicy at all, so no finding from AR014.
function uploadWithRetry(client, path, maxAttempts) {
  let attempts = 0;
  while (attempts < maxAttempts) {
    try {
      client.upload(path);
      break;
    } catch (err) {
      attempts += 1;
    }
  }
}
